### PR TITLE
Shard the processor so it runs on multiple goroutines

### PR DESCRIPTION
**Pull request by stephenmcgruer**

### CODE REVIEW

File: processor/migrate.go [hunk 1]
```diff
@@ -52,28 +78,7 @@ func MigrateData(runsProcessor Runs) {
...
-			_, ok := err.(ConditionUnsatisfied)
-			if !ok {
-				panic(err)
-			} else {
-				continue
-			}
-		}
-		fmt.Printf("Processed TestRun %s (%s %s)\n", key.String(), run.BrowserName, run.BrowserVersion)
+
+		go ProcessRun(ctx, runsProcessor, dsClient, key)
```
Comment (Hexcles) [inline processor/migrate.go]: The main goroutine needs to wait for all other goroutines explicitly.

There are two common approaches: `WaitGroup` or a channel.

**Review by Hexcles — CHANGES_REQUESTED** [review]

File: processor/migrate.go (hunk 1 above)
Comment (stephenmcgruer) [reply processor/migrate.go]: Wouldn't that be sensible. Done.

File: processor/migrate.go [hunk 2]
```diff
@@ -52,28 +82,8 @@ func MigrateData(runsProcessor Runs) {
...
-				panic(err)
-			} else {
-				continue
-			}
-		}
-		fmt.Printf("Processed TestRun %s (%s %s)\n", key.String(), run.BrowserName, run.BrowserVersion)
+
+		wg.Add(1)
+		go ProcessRun(ctx, runsProcessor, dsClient, key, &wg)
 	}
```
Comment (Hexcles) [inline processor/migrate.go:88]:
```suggestion
	}
	wg.Wait()
```

**Review by Hexcles — APPROVED** [review]
LGTM

**Pull request merged**